src/nv_ingest/api/v2/ingest.py

```diff
@@ -0,0 +1,444 @@
+        The PDF file content as bytes
+        
+    Returns
+    -------
+    List[bytes]
+        List of single-page PDFs as bytes
+    """
+    page_pdfs = []
+    
+    # Load the PDF
+    pdf = pdfium.PdfDocument(pdf_content)
+    page_count = len(pdf)
+    
+    for page_index in range(page_count):
+        # Create a new document with just this page
+        new_pdf = pdfium.PdfDocument.new()
+        new_pdf.import_pages(pdf, [page_index])
+        
+        # Save to BytesIO buffer
+        buffer = BytesIO()
+        new_pdf.save(buffer)
+        buffer.seek(0)
+        page_pdfs.append(buffer.read())
+        
+        # Clean up the new PDF
+        new_pdf.close()
+    
+    # Clean up original PDF
+    pdf.close()
+    
+    return page_pdfs
+
+
+def get_pdf_page_count(pdf_content: bytes) -> int:
+    """Get the number of pages in a PDF using pypdfium2."""
+    try:
+        pdf = pdfium.PdfDocument(pdf_content)
+        page_count = len(pdf)
+        pdf.close()
+        return page_count
+    except Exception as e:
+        logger.warning(f"Failed to get PDF page count: {e}")
+        return 1  # Assume single page on error
+
+
+# POST /v2/submit_job
+@router.post(
+    "/submit_job",
+    responses={
+        200: {"description": "Jobs were successfully submitted"},
+        500: {"description": "Error encountered while submitting jobs."},
+        503: {"description": "Service unavailable."},
+    },
+    tags=["Ingestion"],
+    summary="submit jobs to the core nv ingestion service for processing with PDF splitting",
+    operation_id="submit_job_v2",
+)
+async def submit_job_v2(request: Request, response: Response, job_spec: MessageWrapper, ingest_service: INGEST_SERVICE_T):
+    with tracer.start_as_current_span("http-submit-job-v2") as span:
+        try:
+            # Add custom attributes to the span
+            span.set_attribute("http.method", request.method)
+            span.set_attribute("http.url", str(request.url))
+            span.add_event("Submitting file for processing (V2)")
+
+            current_trace_id = span.get_span_context().trace_id
+            parent_job_id = trace_id_to_uuid(current_trace_id)
+
+            # Parse job spec
+            job_spec_dict = json.loads(job_spec.payload)
+            
+            # Extract document type and payload from the proper structure
+            job_payload = job_spec_dict.get("job_payload", {})
+            document_types = job_payload.get("document_type", [])
+            payloads = job_payload.get("content", [])
+            
+            # Check if this is a PDF that needs splitting
+            if document_types and payloads and document_types[0].lower() == "pdf":
+                # Decode the payload to check page count
+                pdf_content = base64.b64decode(payloads[0])
+                page_count = get_pdf_page_count(pdf_content)
+                
+                # Split if more than 3 pages
+                if page_count > 3:
```

Comment:
Why do you need to have this check here? Shouldn't the split logic detect if it needs to split or not?